src/complex_matmul.jl

```diff
@@ -0,0 +1,129 @@
+real_rep(a::AbstractArray{Complex{T}, N}) where {T, N} = reinterpret(reshape, T, a)
+#PtrArray(Ptr{T}(pointer(a)), (StaticInt(2), size(a)...))
+
+@inline function _matmul!(_C::AbstractMatrix{Complex{T}}, _A::AbstractMatrix{Complex{U}}, _B::AbstractMatrix{Complex{V}},
+                         α=One(), β=Zero(), nthread::Nothing=nothing, MKN=nothing, contig_axis=nothing) where {T,U,V}
+    C, A, B =  real_rep.((_C, _A, _B))
+
+    η = IfElse.ifelse(ArrayInterface.is_lazy_conjugate(_A), StaticInt(-1), StaticInt(1))
+    θ = IfElse.ifelse(ArrayInterface.is_lazy_conjugate(_B), StaticInt(-1), StaticInt(1))
+    (+ᶻ, -ᶻ) = IfElse.ifelse(ArrayInterface.is_lazy_conjugate(_C), (-, +), (+, -))
```

Comment:
I haven't added special handling of `StaticInt` or `StaticFloat` to LoopVectorization yet, but it should handle functions like this correctly.
Meaning that, for the time being, choosing functions (like `-ᶻ`) will be friendlier to LoopVectorization's cost modeling than choosing `StaticInt`s.

Reply:
I did some tests where I found that including the `η` and `θ` did not hurt performance on my machine, but doing the same for the `C` array did hurt performance, so I didn't bother making functions for those two.

Could you test and see if just manually deleting the `η` and `θ` has any performance implications on your machine?

Reply:
i.e. try
```julia
Cmn_re +=  A[1, m, k] * B[1, k, n] - A[2, m, k] * B[2, k, n]
Cmn_im += A[1, m, k] * B[2, k, n] + A[2, m, k] * B[1, k, n]
```

Reply:
Assembly doesn't show any problems there, so this is fine.